chain/bitcoind_conn.go

```diff
@@ -437,6 +437,70 @@ func (c *BitcoindConn) GetBlock(hash *chainhash.Hash) (*wire.MsgBlock, error) {
 	}
 }
 
+// GetBlocksBatch returns a batch of raw blocks from the server given their
+// hashes. If the server has already pruned some of the blocks, it will be
+// retrieved from their peers.
+func (c *BitcoindConn) GetBlocksBatch(
+	hashes []*chainhash.Hash) ([]*wire.MsgBlock, error) {
+
+	batchRequest := make([]rpcclient.FutureGetBlockResult, 0, len(hashes))
```

Comment:
Since we have an index here and are not filtering anything, we can pre-allocate with `len(hashes)` as the second parameter directly and then below just use `batchRequest[i] = c.batchclient.GetBlockAsync(hashes[i])`.

Also, I think the plural (`batchRequests`) would make more sense here.